### PR TITLE
feat(ui): discovery ui

### LINKED ISSUE #591: discovery: display available projects
Display the projects that are discoverable at that moment in time.

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] seed config in settings
Screen Shot 2020-07-08 at 2 53 46 PM
Screen Shot 2020-07-08 at 2 53 58 PM

- [x] project discovery ui components
discovery

- [x] polish
- [x] tests (unit & integration)

Closes #591 ; closes ui side of #594

### CODE REVIEW

File: ui/Screen/Discovery.svelte [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  import { Icon, Input, Title } from "../DesignSystem/Primitive";
+
+  import Project from "./Discovery/Project.svelte";
+
+  const projectFeed = getFeed();
+</script>
+
+<style>
+  .container {
+    margin: 64px 100px;
```
Comment (juliendonck) [inline ui/Screen/Discovery.svelte:14]: Wondering if we can use the `--content-padding` variable here.

File: ui/Screen/Discovery.svelte [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  .projects {
+    display: grid;
+    grid-template-columns: minmax(0, 1fr) minmax(0, 1fr);
+    column-gap: 32px;
+    row-gap: 32px;
```
Comment (juliendonck) [inline ui/Screen/Discovery.svelte:21]: I know it's `32px` in the design file, but I think we use `24px` everywhere else. So maybe it's nice to align it. Plus `grid-gap` does the both column and row.
```suggestion
    grid-gap: 24px;
```

File: ui/Screen/Discovery.svelte [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  }
+</style>
+
+<SidebarLayout>
+  <div class="container">
+    <Title variant="huge" style="margin-bottom: 28px;">Discover</Title>
```
Comment (juliendonck) [inline ui/Screen/Discovery.svelte:27]: we try to use multiples of `8px` for everything in the design file. So I'd try `32px` here. (just messy design file)

File: ui/Screen/Discovery.svelte [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    <Input.Text
+      placeholder="Filter projects"
+      showLeftItem
+      style="margin-bottom: 28px; max-width: 350px;">
```
Comment (juliendonck) [inline ui/Screen/Discovery.svelte:31]: use `24px` margin here.

File: ui/Screen/Discovery/Project.svelte [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  const toggleTrackButton = e => {
+    showTrackButton = e.type === "mouseenter";
+  };
+</script>
+
+<style>
+  .container {
+    border: 1px solid var(--color-foreground-level-2);
+    border-radius: 4px;
+    padding: 32px;
```
Comment (juliendonck) [inline ui/Screen/Discovery/Project.svelte:24]: Padding should be `24px` here.

File: ui/Screen/Discovery.svelte (hunk 1 above)
Comment (juliendonck) [reply ui/Screen/Discovery.svelte:14]:
```suggestion
    margin: var(--content-padding) 96px;
```

File: ui/Screen/Settings.svelte [hunk 6]
```diff
@@ -2,52 +2,79 @@
...
+  const updateSeeds = event => {
+    updatePeerConfig(event.target.value);
+  };
 </script>
 
 <style>
   .container {
     max-width: var(--content-max-width);
-    margin: 64px auto;
+    margin: 64px 100px;
```
Comment (juliendonck) [inline ui/Screen/Settings.svelte:32]: This doesn't centre the content anymore. Don't understand what you were trying to do here.

File: ui/Screen/Settings.svelte [hunk 7]
```diff
@@ -2,52 +2,79 @@
...
     align-items: center;
     margin-left: 16px;
   }
+
+  /* TODO(sos): implement responsive margins once we have screen-based layouts */
+  @media (max-width: 1000px) {
+    .container {
+      margin: 64px auto;
+    }
+  }
```
Comment (juliendonck) [inline ui/Screen/Settings.svelte:72]: Does this have to do with [this](https://github.com/radicle-dev/radicle-upstream/pull/629/files#diff-a2343ce5bd1bd043d92dd0abf58e632eR32)

File: ui/Screen/Discovery/Project.svelte [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+    overflow: hidden;
+  }
+
+  .container:hover {
+    border: 2px solid var(--color-foreground-level-3);
+    background: var(--color-foreground-level-1);
+  }
```
Comment (juliendonck) [inline ui/Screen/Discovery/Project.svelte:31]: This makes the UI jump cause it's pushing the content around. 
Check out the `Button.svelte` file to see how we solved this with a combination of `box-shadow` and `border`.

File: ui/Screen/Discovery/Project.svelte [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+  <div class="description">
+    <Text>{project.description}</Text>
+  </div>
+
+  <div class="bottom">
+    <Stats
+      branches={project.stats.branches}
+      commits={project.stats.commits}
+      contributors={project.stats.contributors} />
+    <Avatar avatarFallback={project.maintainers[0].avatar} />
```
Comment (juliendonck) [inline ui/Screen/Discovery/Project.svelte:114]: Should be `circle small` variant. Making it 24px in stead of 32px.

File: ui/Screen/Discovery.svelte (hunk 2 above)
Comment (sarahscott) [reply ui/Screen/Discovery.svelte:21]: nice! makes sense. it's tedious to update figma all the time but it'd be cool if the specs were as close to the app as possible.

File: ui/Screen/Settings.svelte (hunk 7 above)
Comment (sarahscott) [reply ui/Screen/Settings.svelte:72]: Yup - trying to make it look more like the spec, which I actually think looks nicer than the stretched version. I can remove it for now and we can do a responsive overhaul before the release. related to #577

**Comment by sarahscott** [comment]
@juliendonck i can go ahead and fix up the comments you made and remove the changes to the overall settings layout. @xla should we revisit responsiveness after #582 ?  It would be nice to have a unified approach instead of updating each screen individually to accommodate smaller screens.

**Comment by sarahscott** [comment]
@xla & @juliendonck I've addressed your feedback, lgtm?

**Comment by juliendonck** [comment]
I'll try to review it tonight so you're unblocked for your day @sarahscott. It'll be later though. 👋

**Comment by sarahscott** [comment]
@brandonhaslegs thanks for the review!  A few caveats:

1. It was easier to make a `seed` class that resembles the outlined `Button` than to overhaul `Button` layout logic; let me know if it looks off
2. I'm going to do the multiple avatars + tooltip refactor in a separate PR
3. We decided not to do feed-sorting this cycle; once we have the API up and running, we can see what kind of capacity we have for filtering/sorting the feed

**Comment by brandonhaslegs** [comment]
Just realized the Settings page margins are broken in this branch.

### Before:
image

---

### In this branch:
image

File: ui/Screen/Settings.svelte [hunk 10]
```diff
@@ -25,29 +43,80 @@
     min-width: var(--content-min-width);
     padding: 0 var(--content-padding);
   }
+
   section header {
     margin: 16px 0 24px 0;
     border-bottom: 1px solid var(--color-foreground-level-3);
-    padding: 12px;
+    padding: 12px 0;
```
Comment (juliendonck) [inline ui/Screen/Settings.svelte:50]: This breaks the alignment as Brandon illustrated here: https://github.com/radicle-dev/radicle-upstream/pull/629#issuecomment-655998993

File: ui/Screen/Settings.svelte [hunk 11]
```diff
@@ -25,29 +43,80 @@
...
+  section header a {
+    color: var(--color-secondary);
+  }
+
+  section header a:hover {
+    text-decoration: underline;
   }
+
   .section-item {
-    padding: 0 12px;
```
Comment (juliendonck) [inline ui/Screen/Settings.svelte:34]: Same here https://github.com/radicle-dev/radicle-upstream/pull/629#issuecomment-655998993

**Comment by sarahscott** [comment]
oo sorry i think i was going off an old spec or something with the settings side margins. thought that was a bug and not a feature 😂

**Comment by cloudhead** [comment]
Looks great. The only thing I'm wondering is: how do the org links work? Should they only be linked if you are a member of the org, and just be black/unlinked if you arent?

**Comment by sarahscott** [comment]
@cloudhead good question - let's discuss on https://github.com/radicle-dev/radicle-upstream/issues/613

**Pull request merged**